pkg/core/processor/derivproc.go

```diff
@@ -28,11 +26,12 @@ type templateData struct {
 	ReqID  int64
 }
 
-// New creates and returns a new Processor instance configured with the provided Config.
+// NewDeriv creates and returns a new Processor instance configured with the provided Config.
 // It takes a single parameter cfg of type *Config which contains the necessary configuration.
 // It returns a pointer to a Processor struct initialized with the values from the Config.
-func New(cfg *Config) *Processor {
-	return &Processor{
+func NewDeriv(cfg *Config) *DerivProc {
```

Comment:
should `DerivProc` impement `Processor`?
can the return type here be `*Processor`?

Reply:
@shan-96 rule of thumb we should return concrete implementation and accept interfaces.   We return interface in processor factory, but here there is no reason to hide concrete implementation.